src/hooks/useBridgedBlocks.ts

```diff
@@ -36,32 +37,53 @@ interface Props {
 }
 
 const useBridgedBlocks = ({ isApiReady, api, chain }: Props) => {
-  const [bestBridgedFinalizedBlock, setBestBridgedFinalizedBlock] = useState('');
-
+  const [bestBridgedFinalizedBlock, setBestBridgedFinalizedBlock] = useMountedState('');
+  const [bestFinalizedBlock, setBestFinalizedBlock] = useMountedState('');
   const { bridgedGrandpaChain } = getSubstrateDynamicNames(chain);
+
+  useEffect((): (() => void) | undefined => {
+    let unsubBestFinalized: Promise<VoidFn>;
+
+    const shouldProceed: boolean = !!(isApiReady && chain);
+    if (shouldProceed) {
+      unsubBestFinalized = api.query[bridgedGrandpaChain].bestFinalized((res: CodecHeaderId) => {
+        const bestFinalized = res.toString();
+        setBestFinalizedBlock(bestFinalized);
+      });
+    }
+
+    return () => {
+      unsubBestFinalized
+        .then((u) => {
+          u();
+        })
+        .catch((e) => logger.error('error unsubscribing unsubBestFinalized', e));
+    };
+  }, [isApiReady, chain, api, bridgedGrandpaChain, setBestFinalizedBlock]);
```

Comment:
```suggestion
    const shouldProceed: boolean = !!(isApiReady && chain);
    if (!shouldProceed) {
      return;
    }
    
    const unsubBestFinalized = api.query[bridgedGrandpaChain].bestFinalized((res: CodecHeaderId) => {
        const bestFinalized = res.toString();
        setBestFinalizedBlock(bestFinalized);
    });

    return async () => {
      try {
        (await unsubBestFinalized)();
      } catch (e) {
        logger.error('error unsubscribing unsubBestFinalized', e)
      }
     };
  }, [isApiReady, chain, api, bridgedGrandpaChain, setBestFinalizedBlock]);
```

This is imho a bit cleaner, but it's more a matter of preference. Alternatively please just change `then(u => u())` to this more concise form.

Reply:
https://github.com/paritytech/parity-bridges-ui/pull/179/files#r641805104

Reply:
It's not only about `async/await`.

Reply:
please check now.

Reply:
No, it's still broken. if `shouldProceed === false you will call `.then` on `undefined` value in cleanup.

Reply:
not addressed

Reply:
i think this discussion got outdated with PR #181